Gist/Managers/MessageManager.swift

```diff
@@ -77,48 +77,104 @@ class MessageManager: EngineWebDelegate {
+
+                     First, try to open the link inside of the host app. This is to keep compatability with Universal Links.
+                     Learn more of edge case: https://github.com/customerio/customerio-ios/issues/262
+
+                     Fallback to opening the URL through a sytem call if:
+                     1. deep link is an app scheme URL
+                     2. Customer has not implemented the correct function in their host app to handle universal link:
+                     ```
+                     func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool
+                     ```
+                     3. Customer returned `false` from ^^^ function.
+                     */
+                    let handledByUserActivity = continueNSUserActivity(webpageURL: url)
+                    
+                    if !handledByUserActivity {
+                        // If `continueNSUserActivity` could not handle the URL, try opening it directly.
+                        UIApplication.shared.open(url) { handled in
+                            if handled {
+                                Logger.instance.info(message: "Dismissing from system action: \(action)")
+                                self.dismissMessage()
+                            } else {
+                                Logger.instance.info(message: "System action not handled")
+                            }
                         }
+                    } else {
+                        Logger.instance.info(message: "Handled by NSUserActivity")
+                        self.dismissMessage()
                     }
                 }
             }
+            
         }
     }
+    
+    // Check if
+    func continueNSUserActivity(webpageURL: URL) -> Bool {
+        guard #available(iOS 10.0, *) else {
+            return false
+        }
+        guard isLinkValidNSUserActivityLink(webpageURL) else {
+            return false
+        }
+        
+        let openLinkInHostAppActivity = NSUserActivity(activityType: NSUserActivityTypeBrowsingWeb)
+        openLinkInHostAppActivity.webpageURL = webpageURL
+        
+        let didHostAppHandleLink = UIApplication.shared.delegate?.application?(UIApplication.shared, continue: openLinkInHostAppActivity, restorationHandler: { _ in }) ?? false
+
+        return didHostAppHandleLink
+    }
+        
+        
+    // The NSUserActivity.webpageURL property permits only specific URL schemes. This function exists to validate the scheme and prevent potential exceptions due to incompatible URL formats.
+    func isLinkValidNSUserActivityLink(_ url: URL) -> Bool {
+        guard let schemeOfUrl = url.scheme else {
+            return false
+        }
+        
+        // Constants to hold allowed URL schemes
+        let allowedSchemes = ["http", "https"]
+
+        return allowedSchemes.contains(schemeOfUrl)
+    }
```

Comment:
This code is similar to [`UIKitWrapper` in the iOS SDK](https://github.com/customerio/customerio-ios/blob/main/Sources/Common/Util/UIKitWrapper.swift) code. 

Because we plan on moving the Gist SDK into the iOS SDK soon, we will be able to re-use the code that is already present in the iOS SDK. 

With that in mind, I suggest that we copy/paste the `UIKitWrapper` source code file from the iOS SDK into the Gist SDK, replace this code with a call to the `UIKitWrapper`, and then when we merge the Gist SDK into the iOS SDK, we will be able to delete the duplicate `UIKitWrapper` file. 

I am open to ideas if that does not work. I want to avoid what is in the PR right now where there is similar code copied into 1 file.

Reply:
My suggestion of having a copy/pasted file is that it would be a smaller amount of work to be done when we perform this future refactor. When we do a refactor, we could add delete the duplicate `UIKitWrapper` file, add a Swift `import CioCommon` to the Gist file, then the refactor is complete. 

Taking a step back for this suggestion, I am trying to avoid us forgetting to perform the refactor of removing this duplicate code introduced in this PR. 

I would be OK with either creating a ticket now that documents this refactor or having a copy/pasted file with a comment on it that the file will be deleted in the future.

Reply:
Created the [ticket](https://github.com/customerio/issues/issues/10294).